src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -0,0 +1,392 @@
+import java.io.IOException;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.AccessDeniedException;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.Set;
+
+import com.synopsys.integration.blackduck.service.BlackDuckServicesFactory;
+import com.synopsys.integration.detect.configuration.DetectInfo;
+import com.synopsys.integration.detect.configuration.DetectInfoUtility;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.log.LogLevel;
+import com.synopsys.integration.log.SilentIntLogger;
+import com.synopsys.integration.rest.HttpMethod;
+import com.synopsys.integration.rest.HttpUrl;
+import com.synopsys.integration.rest.client.IntHttpClient;
+import com.synopsys.integration.rest.proxy.ProxyInfo;
+import com.synopsys.integration.rest.request.Request;
+import com.synopsys.integration.rest.response.Response;
+
+import freemarker.template.Version;
+import java.util.LinkedList;
+import org.apache.commons.lang3.StringUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ApplicationUpdater extends URLClassLoader {
+    
+    public static final String DOWNLOAD_VERSION_HEADER = "Version";
+    private static final String LOG_PREFIX = "Detect-Self-Updater: ";
+    private static final String DOWNLOAD_URL = "api/tools/detect";
+    private static final String DOWNLOADED_FILE_NAME = "X-Artifactory-Filename";
+    private static final Version MINIMUM_DETECT_VERSION = new Version(8, 9, 0);
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final IntHttpClient intHttpClient;
+    private final String[] args;
+    private String blackduckHost = null;
+    private String offlineMode = null;
+    private final DetectInfo detectInfo;
+    
+    private final static String DETECT_SOURCE_SYS_ENV_PROP = "DETECT_SOURCE";
+    private final static String DETECT_LATEST_RELEASE_VERSION_SYS_ENV_PROP = "DETECT_LATEST_RELEASE_VERSION";
+    private final static String DETECT_VERSION_KEY_SYS_ENV_PROP = "DETECT_VERSION_KEY";
+    private final static String BLACKDUCK_OFFLINE_MODE_ARG = "blackduck.offline.mode";
+    private final static String BLACKDUCK_URL_ARG = "blackduck.url";
+    
+    public ApplicationUpdater(String[] args) {
+        super(new URL[] {}, Thread.currentThread().getContextClassLoader());
+        this.args = parseArguments(args);
+        final SilentIntLogger silentLogger = new SilentIntLogger();
+        silentLogger.setLogLevel(LogLevel.WARN);
```

Comment:
The xml is global for httpclient logger name of the org.apache wire log. The silent logger WARN here is for IntHttpClient flow which otherwise prints way too many INFOs.